### ISSUE TITLE
TypeError: Cannot read property 'contentType' of undefined

### ISSUE THREAD

**Issue by doobnet** [slimapi/apib2json#7]
I tried to run apib2json on the following Blueprint: https://airdcpp.docs.apiary.io. I get the following error:

```
/src/apib2json/lib/apib2json.js:92
                                        if (isAsset && rContent.attributes.contentType === 'application/schema+json') {
                                                                           ^

TypeError: Cannot read property 'contentType' of undefined
    at httpTransaction.content.forEach (/src/apib2json/lib/apib2json.js:92:76)
    at Array.forEach (<anonymous>)
    at content.content.forEach (/src/apib2json/lib/apib2json.js:90:61)
    at Array.forEach (<anonymous>)
    at transition.content.forEach (/src/apib2json/lib/apib2json.js:72:49)
    at Array.forEach (<anonymous>)
    at resource.content.forEach (/src/apib2json/lib/apib2json.js:66:48)
    at Array.forEach (<anonymous>)
    at group.content.forEach (/src/apib2json/lib/apib2json.js:61:42)
    at Array.forEach (<anonymous>)
```

I did some debugging and there seems to be two resources that are missing the `attributes` key:

* https://airdcpp.docs.apiary.io/#reference/filelist-entities/methods/get-directory-content
* https://airdcpp.docs.apiary.io/#reference/share/generic-methods/validate-path

It seems to be when the response code is anything other than 2xx.

**Comment by visch** [comment]
Same with https://jsapi.apiary.io/apis/clickup20.source

**Comment by o5** [comment]
@visch Thanks for you report! I fixed it in referenced commit, you can try it `slimapi/apib2json:0.3.0-4-g8ee2b48`.

Your linked API Blueprint does not have [Attributes section](https://apiblueprint.org/documentation/specification.html#def-attributes-section) nor attached JSON Schema(s) so this tool not really help you. You can look [into fixtures](https://github.com/slimapi/apib2json/blame/master/test/fixtures/advanced_attributes_api.apib.md#L38-L45) how to write attributes and [how output](https://github.com/slimapi/apib2json/blob/master/test/fixtures/advanced_attributes_api.expected.json#L50-L83) looks like. The main job of this tool is to get the content of **SHOW JSON SCHEMA** link in Apiary Editor as you can see in my screens.

![Screenshot 2021-09-04 at 0 12 48](https://user-images.githubusercontent.com/218562/132070869-a344602e-166e-49b5-87c5-065afa047ade.png)
![Screenshot 2021-09-04 at 0 12 18](https://user-images.githubusercontent.com/218562/132070870-12cf3dc4-692c-47a3-a412-e7b10154cb50.png)

**Comment by visch** [comment]
Tried to do this had multiple issues (not an npm developer) 

1. Started with the node docker image. https://hub.docker.com/_/node , moved to a directory, cloned down the repo, and couldn't get `npm install slimapi/apib2json:0.3.0-4-g8ee2b48` to work
2. Tried getting to the commit locally, 1. `git clone git@github.com:slimapi/apib2json.git` , 

![image](https://user-images.githubusercontent.com/8680264/132260062-2f382856-080f-4fcc-aef0-d24541116e8f.png)

**Comment by o5** [comment]
Just

```bash
docker run --rm -i slimapi/apib2json:0.3.0-4-g8ee2b48 --pretty < documentation.md > output.new.json
```

**Comment by visch** [comment]
Ok I get it now I think, when you say `slimapi/apib2json:0.3.0-4-g8ee2b48` you're referring to the dockerhub tag. When I saw the hash `g8ee2b48` I assumed a git hash and tried to pull the git hash. So with these dev builds there's not a git hash alignment with dockerhub? It's a bit scary from my end running code that I can't see the source, sharing this as I'd want to know if I was the maintainer. This slowed me down from using this project (if you care or not I don't know just wanted to share). 

I'll give this a shot in an isolated vm (tin foil hat man here)

**Comment by o5** [comment]
Sorry for misunderstanding! Short hash `g8ee2b48` is result from [`git describe`](https://github.com/slimapi/apib2json/blob/updates/.github/workflows/ci.yml#L25) and it is really short GIT hash but there is also prefix `g`.

From [GIT docs](https://git-scm.com/docs/git-describe#_examples):
> The "g" prefix stands for "git" and is used to allow describing the version of a software depending on the SCM the software is managed with. This is useful in an environment where people may use different SCMs

Unfortunately, there is a development branch `updates` where I currently maintains big changes not only for application itself but also for new CI (Travis -> GitHub Actions). GA is almost impossible to setup with only one commit so I used force push in this branch. I wanted to merge to the master after [someone's check](https://github.com/slimapi/apib2json/pull/9#issuecomment-912809417) but he probably not have time. But main fact is: I transferred this repo to new namespace (from personal `o5` to `slimapi`). NPM is still pointed to old, so for now, that version is not possible to load over NPM.

Lessons learned: It is not a good idea to prepare this type of changes in "one" step :) I would like to release the stable on this weekend, stay tuned.
